test/Router.spec.js

```diff
@@ -5,6 +5,19 @@ import HistoryEnv from '../src/HistoryEnv'
 import DocumentEnv from '../src/DocumentEnv'
 import fns from '../src/fns'
 
+/**
+ * Helper function to generate a Promise that is resolved
+ * after a setTimeout (macro-task timing).
+ * This is useful to control test timing where our
+ * router resolves routes using Promises (micro-task timing)
+ * More on micro-tasks vs macro-tasks:
+ *   https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/
```

Comment:
very interesting read thanks!

Reply:
NP! See here for where Nikhil and I learned about this timing behavior.

https://github.com/optimizely/client-js/blob/devel/src/plugins/change_attribute/index.js#L175-L196